client/src/schema/schema.ts

```diff
@@ -8498,11 +8498,15 @@ export interface operations {
             /** @description If non-null, get the specified filename from the extra files for this dataset. */
             /** @description The file extension when downloading the display data. Use the value `data` to let the server infer it from the data type. */
             /** @description The query parameter 'raw' should be considered experimental and may be dropped at some point in the future without warning. Generally, data should be processed by its datatype prior to display. */
+            /** @description Set this for datatypes that allow chunked display through the display_data method to enable chunking. This specifies a byte offset into the target dataset's display. */
+            /** @description If offset is set, this recommends a number of chunked lines to output from the display API. This is not respected uniformly and should be interpreted as a recommendation. For instance, with bam datatypes this is ignored for the header lines but the remainder of the file respects this parameter. */
```

Comment:
I've been wondering about exactly this definition when I touched the implementation for bam files last time.
While it's true that for BAMs, ck_size is interpreted as the number of lines in a chunk, for TabularData, for example, it seems to have a rather different meaning: it specifies the minimal number of bytes to display, which are then extended up to the next line ending (see https://github.com/galaxyproject/galaxy/blob/923b5363f830b19ea71add72dff45e9eab56a7bd/lib/galaxy/datatypes/tabular.py#L145).

I guess the byte-based ck_size makes more sense, but have been wondering, which one is the intended behavior.
If BAMs are supposed to behave like tabular data, I'd be happy to rewrite this once more.

Reply:
I pushed an update that explains this situation in a little more detail. I don't find this discrepancy problematic enough to warrant asking you to spend time fixing it but I try not to get hung up on consistency. If it is important to you feel free to pick a consistent definition for this parameter and unify the approaches - or something else like splitting this into ``chunk_size_bytes`` and ``chunk_size_lines``. I don't think this API has an external consumer so I wouldn't worry about backward compatibility as long as the UI works.